docs/source/notes/amp_examples.rst

```diff
@@ -0,0 +1,210 @@
+        grad_norm = grad_norm.sqrt()
+        loss = loss + grad_norm
+
+        # The usual scaling for backward will now accumulate leaf gradients that are appropriately scaled.
+        scaler.scale(loss).backward()
+        scaler.step(optimizer)
+        scaler.update()
+
+
+Gradient accumulation
+---------------------
+
+Gradient accumulation across iterations (between steps) is a common use case.
+:class:`AmpScaler` accommodates gradient accumulation without trouble::
+
+    scaler = AmpScaler()
+    ...
+    for i, (input, target) in enumerate(data):
+        output = model(input)
+        loss = loss_fn(output, target)
+        loss = loss/iters_to_accumulate
+        scaler.scale(loss).backward()
+        if (i + 1) % iters_to_accumulate == 0:
+            scaler.step(optimizer)
+            scaler.update()
+            optimizer.zero_grad()
+
+Switching gradient scaling on and off
+------------------------------------
+
+The ``enabled`` kwarg to :class:`AmpScaler` allows gradient scaling to be globally enabled/disabled without script-side if statements::
+
+    scaler = AmpScaler(enabled=args.use_mixed_precision)
+    ...
+    for input, target in data:
+        optimizer.zero_grad()
+        output = model(input)
+        loss = loss_fn(output, target)
+        scaler.scale(loss).backward()
+        scaler.step(optimizer)
+        scaler.update()
+
+If ``enabled=False``, ``scaler.step(optimizer)`` directly invokes ``optimizer.step()`` without any wrapping logic, and
+the other methods (``scaler.scale``, ``scaler.update``) become no-ops.
+
+
+Multiple models/optimizers/losses
+---------------------------------
+
+Make sure to call ``scaler.update()`` only at the end of the iteration, after ``scaler.step(optimizer)`` has
+been called for all optimizers used this iteration::
+
+    scaler = torch.cuda.amp.AmpScaler()
+
+    for input, target in data:
+        optimizer0.zero_grad()
+        optimizer1.zero_grad()
+        output0 = model0(input)
+        output1 = model1(input)
+        loss0 = loss_fn(2 * output0 + 3 * output1, target)
+        loss1 = loss_fn(3 * output0 - 5 * output1, target)
+
+        scaler.scale(loss0).backward(retain_graph=True)
+        scaler.scale(loss1).backward()
+
+        # You can choose which optimizers receive explicit unscaling
+        scaler.unscale(optimizer0)
+
+        scaler.step(optimizer0)
+        scaler.step(optimizer1)
+        scaler.update()
+
+Note that the decision to invoke an explicit ``unscale`` can be made independently for each optimizer.
+
+.. warning::
+    If you (optionally) choose to unscale gradients prior to stepping, ``scaler.unscale(optimizer)``
+    should only be invoked once per optimizer per step, and only after all gradients for that optimizer's
+    owned parameters have been accumulated.
+
+.. _custom-optimizer-guide:
+
+Custom Optimizer Guide
+^^^^^^^^^^^^^^^^^^^^^^
+
+I will write this once the gradient scaling API is approved.
```

Comment:
Need to remove or update this. I think you want to remove it. There can be a note about an extensibility point per comment above but I wouldn't elaborate.

Reply:
Should I delete references to custom optimizer handling to simplify this PR and add them later?

The important thing is that right now we are satisfied with the contract that custom optimizers may opt into.  I think it's fine but we should review it.  If it's fine, then any later changes will be pure documentation.